panther_analysis_tool/rule.py

```diff
@@ -79,27 +95,35 @@ class RuleResult:
 
     dedup_output: Optional[str] = None
     dedup_exception: Optional[Exception] = None
+    dedup_defined: bool = False
 
     title_output: Optional[str] = None
     title_exception: Optional[Exception] = None
+    title_defined: bool = False
 
     description_output: Optional[str] = None
     description_exception: Optional[Exception] = None
+    description_defined: bool = False
 
     reference_output: Optional[str] = None
     reference_exception: Optional[Exception] = None
+    reference_defined: bool = False
 
     severity_output: Optional[str] = None
     severity_exception: Optional[Exception] = None
+    severity_defined: bool = False
 
     runbook_output: Optional[str] = None
     runbook_exception: Optional[Exception] = None
+    runbook_defined: bool = False
 
     destinations_output: Optional[List[str]] = None
     destinations_exception: Optional[Exception] = None
+    destinations_defined: bool = False
 
-    alert_context: Optional[str] = None
+    alert_context_output: Optional[str] = None
```

Comment:
Renamed for uniformity and so that we can fetch output dynamically by function name.